### PR TITLE
backport: merge bitcoin#15508, #18764, #19264, #19107, #17577, #18783, #15024, #18204, partial #14588, #13932 (auxiliary backports: part 6)

### PR DESCRIPTION
**Pull request by kittywhiskers**
## Issue being fixed or feature implemented
Backports that are a prerequisite for https://github.com/dashpay/dash/pull/5000

## What was done?
See commit log for bitcoin/bitcoin pull requests that were merged entirely or partially.

## How Has This Been Tested?
Compilation, running unit and functional tests

## Breaking Changes
Adds new RPC call `analyzepsbt`

## Checklist:
* [x]  I have performed a self-review of my own code
* [x]  I have commented my code, particularly in hard-to-understand areas
* [x]  I have added or updated relevant unit/integration/functional/e2e tests
* [x]  I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**

* [x]  I have assigned this pull request to a milestone

### CODE REVIEW

File: src/rpc/rawtransaction.cpp [hunk 1]
```diff
@@ -1575,6 +1575,197 @@ UniValue joinpsbts(const JSONRPCRequest& request)
     return EncodeBase64(ssTx.str());
 }
 
+UniValue analyzepsbt(const JSONRPCRequest& request)
```
Comment (PastaPastaPasta) [inline src/rpc/rawtransaction.cpp:1578]: 13932 needs release notes

Comment (kittywhiskers) [reply src/rpc/rawtransaction.cpp:1578]: Done

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
👍 
few things I noticed after a quick review:
1. e6616bd backports 14588, not 15508
2. missing changes? (see below)

File: src/wallet/psbtwallet.cpp [hunk 2]
```diff
@@ -27,7 +27,7 @@ TransactionError FillPSBT(const CWallet* pwallet, PartiallySignedTransaction& ps
             return TransactionError::SIGHASH_MISMATCH;
         }
 
-        complete &= SignPSBTInput(HidingSigningProvider(pwallet->GetSigningProvider(), !sign, !bip32derivs), *psbtx.tx, input, i, sighash_type);
+        complete &= SignPSBTInput(HidingSigningProvider(pwallet->GetSigningProvider(), !sign, !bip32derivs), psbtx, i, sighash_type);
```
[the remark this answers is not in the document]
Comment (kittywhiskers) [reply src/wallet/psbtwallet.cpp:39]: Resolved in latest push

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**